pennylane/decomposition/decomposition_graph.py

```diff
@@ -55,7 +55,19 @@
 from .utils import translate_op_alias
 
 
-class DecompositionGraph:  # pylint: disable=too-many-instance-attributes
+@dataclass(frozen=True)
+class _DecompositionNode:
```

Comment:
We usually add private/protected classes/methods after public ones. Any reasons behind this relocation?

Reply:
I prefer to organize my code following the flow of logic. It makes more sense to define the nodes and edges of a graph before defining the graph itself. This way if you read it from top to bottom, you don't have to jump to the end of the file to figure out what a node is.